app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -0,0 +1,161 @@
+package fr.free.nrw.commons.utils
+
+import android.app.Activity
+import android.app.AlertDialog
+import android.app.Dialog
+import android.view.View
+import fr.free.nrw.commons.R
+import timber.log.Timber
+
+object DialogUtil {
+    /**
+     * Shows a dialog safely.
+     * @param activity the activity
+     * @param dialog the dialog to be shown
+     */
+    private fun showSafely(activity: Activity?, dialog: Dialog?) {
+
+        if (activity == null || dialog == null) {
+            Timber.d("Show called with null activity / dialog. Ignoring.")
+            return
+        }
+
+        if (activity.isFinishing || activity.isDestroyed) {
+            Timber.e("Activity is not running. Could not show dialog. ")
+            return
+        }
+        try {
+            dialog.show()
+        } catch (e: IllegalStateException) {
+            Timber.e(e, "Could not show dialog.")
+        }
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?
+    ) {
+        createAndShowDialogSafely(
+            activity = activity,
+            title = title,
+            message = message,
+            positiveButtonText = activity.getString(R.string.yes),
+            negativeButtonText = activity.getString(R.string.no),
+            onPositiveBtnClick = onPositiveBtnClick,
+            onNegativeBtnClick = onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String,
+        negativeButtonText: String,
```

Comment:
null is passed here which would cause a crash

Reply:
![image](https://user-images.githubusercontent.com/3358282/79216807-b56ec480-7e45-11ea-8898-fcdf8c6c1cf7.png)

Please review the call sites and change the nullability of parameters to fit the use case

Reply:
Done!

Reply:
The nullability is still not really where I want it, take for example
```
    @JvmStatic
    fun showAlertDialog(
        activity: Activity,
        title: String,
        message: String,
        onPositiveBtnClick: Runnable?,
        onNegativeBtnClick: Runnable?
    ) 
```

and its 1 usage

```
DialogUtil.showAlertDialog(getActivity(),
                getString(R.string.upload_nearby_place_found_title),
                String.format(Locale.getDefault(),
                        getString(R.string.upload_nearby_place_found_description),
                        place.getName()),
                () -> {
                    etTitle.setText(place.getName());
                    Description description = new Description();
                    description.setLanguageCode("en");
                    description.setDescriptionText(place.getLongDescription());
                    descriptions = Arrays.asList(description);
                    setDescriptionsInAdapter(descriptions);
                },
                () -> {

                });
```

`onPositiveBtnClick: Runnable?` this is not true because this is never null
`onNegativeBtnClick: Runnable?` this parameter shouldn't even exist


I suppose this is getting a bit more than "conversion" if we are altering the callsites overmuch but I want to hammer home the point that making non nullable things nullable weakens our knowledge of the system from a type perspective. This method's intention is
 "show a Yes/No Dialog with a positive action", to make the action nullable feels awkward. If this method were used somewhere else with a null positive action then I would be okay with it because it would then be "show a Yes/No Dialog with an optional positive action" but the definition and the usage are at odds with one another.

Anyway enough rambling, I am good to merge.